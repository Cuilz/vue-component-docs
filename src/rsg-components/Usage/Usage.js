import React from 'react';
import PropTypes from 'prop-types';
import Props from 'rsg-components/Props';
import Methods from 'rsg-components/Methods';
import Events from 'rsg-components/Events';
import SlotsTable from 'rsg-components/SlotsTable';
import pickBy from 'lodash/pickBy';

export default function Usage({ props: { props, methods, events, slots } }) {
	let slotsNode;
	slots = pickBy(slots, item => !!item.description);
	if (slots && Object.keys(slots).length > 0) {
		slotsNode = slots && <SlotsTable props={slots} />;
	}
	const propsNode = props && <Props props={props} />;
	let eventsNode;
	if (events && Object.keys(events).length > 0) {
		eventsNode = events && <Events props={events} />;
	}
	const methodsNode = methods && methods.length > 0 && <Methods methods={methods} />;

	if (!propsNode && !methodsNode && !slotsNode && !eventsNode) {
		return null;
	}

	return (
		<div>
			{propsNode}
			{methodsNode}
			{eventsNode}
			{slotsNode}
		</div>
	);
}

Usage.propTypes = {
	props: PropTypes.shape({
		props: PropTypes.array,
		methods: PropTypes.array,
		eventsNode: PropTypes.object,
		slotsNode: PropTypes.object,
	}).isRequired,
};
