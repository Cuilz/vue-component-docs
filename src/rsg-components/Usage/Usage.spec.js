import React from 'react';
import Usage from './Usage';

const props = [
	{
		name: 'children',
		type: { name: 'string' },
		required: true,
		description: 'Button label.',
	},
];
const methods = [
	{
		name: 'set',
		params: [
			{
				name: 'newValue',
				description: 'New value for the counter.',
				type: { name: 'Number' },
			},
		],
		returns: null,
		description: 'Sets the counter to a particular value.',
	},
];

describe('Usage', () => {
	it('should render props table', () => {
		const actual = shallow(<Usage props={{ props }} />);

		expect(actual).toMatchSnapshot();
	});

	it('should render methods table', () => {
		const actual = shallow(<Usage props={{ methods }} />);

		expect(actual).toMatchSnapshot();
	});

	it('should render nothing without props and methods', () => {
		const actual = shallow(<Usage props={{}} />);

		expect(actual.getElement()).toBe(null);
	});
});
