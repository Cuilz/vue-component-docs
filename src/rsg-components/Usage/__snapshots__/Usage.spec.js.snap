// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usage should render methods table 1`] = `
<div>
  <MethodsRenderer
    methods={
      Array [
        Object {
          "description": "Sets the counter to a particular value.",
          "name": "set",
          "params": Array [
            Object {
              "description": "New value for the counter.",
              "name": "newValue",
              "type": Object {
                "name": "Number",
              },
            },
          ],
          "returns": null,
        },
      ]
    }
  />
</div>
`;

exports[`Usage should render props table 1`] = `
<div>
  <PropsRenderer
    props={
      Array [
        Object {
          "description": "Button label.",
          "name": "children",
          "required": true,
          "type": Object {
            "name": "string",
          },
        },
      ]
    }
  />
</div>
`;
